added verify contract for etherscan for nft-collection

chee-chyuan opened the pull request:
adding steps to verify contract for etherscan for nft-collection
i will seperately pr for the nft-game after this pr is merged

adilanchian commented on NFT_Collection/en/Section_1/Lesson_4_Create_A_Contract_That_Mints_NFTs.md:
```diff
@@ -329,6 +329,85 @@ Get used to using Etherscan because it's like the easiest way to track deploymen
 
 If it worked — AWEEEEESOME YOU JUST DEPLOYED A CONTRACT YESSSS.
 
+📝Verify contract on Etherscan
```
Can you add a period at the end of this please :)?

adilanchian commented on NFT_Collection/en/Section_1/Lesson_4_Create_A_Contract_That_Mints_NFTs.md:
```diff
@@ -329,6 +329,85 @@ Get used to using Etherscan because it's like the easiest way to track deploymen
+
+It turns out that this long gibberish looking group of characters is actually the bytecodes of the contract which you have deployed! Bytecodes represents a series of opcodes in the EVM that will perform instructions for us onchain. The logic which you have written in Solidity will compile down to bytecodes which the EVM will understand. More about EVM and opcodes [here](https://ethervm.io/).
+
+However bytecodes aren't really readible but Etherscan has the magic to 'convert' them to something more user friendly. For that magic to work, Etherscan will need your help.
+
```
Can you tweak this a bit - 

However bytecodes aren't really readable, but Etherscan has the magic to 'convert' them to something more user friendly. For that magic to work, Etherscan will need your help.

adilanchian commented on NFT_Collection/en/Section_1/Lesson_4_Create_A_Contract_That_Mints_NFTs.md:
```diff
@@ -329,6 +329,85 @@ Get used to using Etherscan because it's like the easiest way to track deploymen
+
+However bytecodes aren't really readible but Etherscan has the magic to 'convert' them to something more user friendly. For that magic to work, Etherscan will need your help.
+
+Notice that there is a prompt that requests us to <b>Verify and Publish</b> our contract source code. If we follow the link, we are required to manually select our contract settings and paste our code to publish our source code.
```
Can you change `<b>Verify and Publish</b>` to use asterisks please `**Verify and Publish**`

adilanchian commented on NFT_Collection/en/Section_1/Lesson_4_Create_A_Contract_That_Mints_NFTs.md:
````diff
@@ -329,6 +329,85 @@ Get used to using Etherscan because it's like the easiest way to track deploymen
+```
+require("@nomiclabs/hardhat-etherscan");
+
+//rest of the code
````
Can you tweak `//rest of code` to be `// Rest of code`

adilanchian commented on NFT_Collection/en/Section_1/Lesson_4_Create_A_Contract_That_Mints_NFTs.md:
```diff
@@ -329,6 +329,85 @@ Get used to using Etherscan because it's like the easiest way to track deploymen
+require("@nomiclabs/hardhat-etherscan");
+
+//rest of the code
+.......
```
Can you change this to `...` please?

adilanchian commented on NFT_Collection/en/Section_1/Lesson_4_Create_A_Contract_That_Mints_NFTs.md:
```diff
@@ -329,6 +329,85 @@ Get used to using Etherscan because it's like the easiest way to track deploymen
+module.exports = {
+  solidity: "0.8.0",
+
+  //rest of the config
+  ....,
```
Can you change this to:

`// Rest of the config`

`...`

adilanchian commented on NFT_Collection/en/Section_1/Lesson_4_Create_A_Contract_That_Mints_NFTs.md:
```diff
@@ -329,6 +329,85 @@ Get used to using Etherscan because it's like the easiest way to track deploymen
+    // Obtain one at https://etherscan.io/
+    apiKey: "",
+  }
+
```
Remove this space please :)

adilanchian commented on NFT_Collection/en/Section_1/Lesson_4_Create_A_Contract_That_Mints_NFTs.md:
```diff
@@ -329,6 +329,85 @@ Get used to using Etherscan because it's like the easiest way to track deploymen
+  }
+
+};
+
```
Remove this space please :)

adilanchian commented on NFT_Collection/en/Section_1/Lesson_4_Create_A_Contract_That_Mints_NFTs.md:
````diff
@@ -329,6 +329,85 @@ Get used to using Etherscan because it's like the easiest way to track deploymen
+
+```
+
+Not quite done yet. Those with keen eyes will notice that Etherscan requires an ApiKey. 
````
Can you change this to say - 

We are almost there! You may have noticed that the `etherscan` object in our config requires an `apiKey`! This means you will need an account with Etherscan to get this key.

adilanchian commented on NFT_Collection/en/Section_1/Lesson_4_Create_A_Contract_That_Mints_NFTs.md:
```diff
@@ -329,6 +329,85 @@ Get used to using Etherscan because it's like the easiest way to track deploymen
+
+Not quite done yet. Those with keen eyes will notice that Etherscan requires an ApiKey. 
+
+If you don't have an account already, head to etherscan.io to create a free user account. After which head to your profile settings and under `API-KEYs` create a new apikey 
```
Can you change this to include the etherscan.io signup link?

adilanchian commented on NFT_Collection/en/Section_1/Lesson_4_Create_A_Contract_That_Mints_NFTs.md:
```diff
@@ -329,6 +329,85 @@ Get used to using Etherscan because it's like the easiest way to track deploymen
+
+![image](https://user-images.githubusercontent.com/60590919/139610459-b590bbc1-0d4e-4e78-920b-c45e61bf2d7e.png)
+
+After which, add your api key to the etherscan apikey property in our `hardhat.config.js` similarly we will be using `.env` later on for this.
```
Can you tweak this to say - 

Sweet you got your API key. Time to head back to your `hardhat.config.js` file and change the `apiKey` property to be your newly generated key.

adilanchian commented on NFT_Collection/en/Section_1/Lesson_4_Create_A_Contract_That_Mints_NFTs.md:
```diff
@@ -329,6 +329,85 @@ Get used to using Etherscan because it's like the easiest way to track deploymen
+
+![image](https://user-images.githubusercontent.com/60590919/139611805-b2a41039-ec79-402d-b198-4936d25ff277.png)
+
+Read contract doesn't change the state of the chain and so its free to use. Write contract however do, and so when interacting with it, you will need to pay gas. But its on the Rinkeby test net, so feel free to request more free Rinkeby Test Eth from the Rinkeby faucet so you have enough to play with.
+
```
I don't think we actually need this part :). Can you remove please?

chee-chyuan commented:
> Hey @chee-chyuan !
> 
> This is an AWESOME addition to this project. I actually learned some new stuff here myself haha. I do think this should actually live [in this section](https://github.com/buildspace/buildspace-projects/blob/main/NFT_Collection/en/Section_2/Lesson_2_Minting_NFTs_With_Static_SVGs.md#-change-up-our-contract-deploy)
> 
> It can be an "extra" instead of a requirement for the project if this makes sense!

Sure thing. I have moved it over to the end of Section 2: lesson 2. and have added the word `Something Extra:` lmk if you would like them removed

chee-chyuan commented:
> Alrighty -
> 
> I left a solid amount of changes here! Overall, your content is great! I just wanted to change a few things up here :)
> 
> Let me know if you have any questions or concerns!

wow thanks! i appreciate your feedback. yours feel much more natural. have edited based on your feedbacks

adilanchian commented on NFT_Collection/en/Section_2/Lesson_2_Minting_NFTs_With_Static_SVGs.md:
```diff
@@ -112,6 +112,84 @@ Note: Remember to use `https://rinkeby.rarible.com/token/INSERT_DEPLOY_CONTRACT_
+
+📝Something Extra: Verify contract on Etherscan.
+------------------
+If you select the **Contract** tab in Etherscan, you will notice a long list of text characters that starts from `0x608060405234801...` Hmm.. what could that be 🤔 ?
+
```
I think here we might want to start with something like:

**Note: This section is totally optional, but is a really cool part about Etherscan that you should understand.**

If you select the **Contract** tab in Etherscan, you will notice a long list of text characters that starts from `0x608060405234801...` Hmm.. what could that be 🤔 ?

chee-chyuan replied:
gotcha let me quicky add that in too

adilanchian commented on NFT_Collection/en/Section_2/Lesson_2_Minting_NFTs_With_Static_SVGs.md:
```diff
@@ -112,6 +112,84 @@ Note: Remember to use `https://rinkeby.rarible.com/token/INSERT_DEPLOY_CONTRACT_
+
+It turns out that this long, gibberish looking group of characters is actually the bytecodes of the contract which you have deployed! Bytecodes represent a series of opcodes in the EVM that will perform instructions for us onchain.
+
+This is a lot of new information to understand, so don't worry if it doesn't make much sense right now. Take a moment to look up what bytecodes and EVM mean! Use Google or reach out in the `#general-chill-chat` on Discord :). [This is also a cool article](https://ethervm.io/) about EVM opcodes
+
```
Can you change `[This is also a cool article](https://ethervm.io/) about EVM opcodes` to - 

[This is also a cool article](https://ethervm.io/) about EVM opcodes by the way 🤘.

adilanchian commented on NFT_Collection/en/Section_2/Lesson_2_Minting_NFTs_With_Static_SVGs.md:
```diff
@@ -112,6 +112,84 @@ Note: Remember to use `https://rinkeby.rarible.com/token/INSERT_DEPLOY_CONTRACT_
+
+This is a lot of new information to understand, so don't worry if it doesn't make much sense right now. Take a moment to look up what bytecodes and EVM mean! Use Google or reach out in the `#general-chill-chat` on Discord :). [This is also a cool article](https://ethervm.io/) about EVM opcodes
+
+However bytecodes aren't really readable, but Etherscan has the magic to 'convert' them to something more user friendly. For that magic to work, Etherscan will need your help.
+
```
Can you change this to - 

So, we know that bytecodes aren't readable to us. We want to be able to see the code we wrote right in Etherscan. Luckily, Etherscan has the magic to help us do that!

adilanchian commented on NFT_Collection/en/Section_2/Lesson_2_Minting_NFTs_With_Static_SVGs.md:
```diff
@@ -112,6 +112,84 @@ Note: Remember to use `https://rinkeby.rarible.com/token/INSERT_DEPLOY_CONTRACT_
+
+We are almost there! You may have noticed that the `etherscan` object in our config requires an `apiKey`! This means you will need an account with Etherscan to get this key.
+
+If you don't have an account already, head to https://etherscan.io/register to create a free user account. After which head to your profile settings and under `API-KEYs` create a new apikey 
```
Can you change `https://etherscan.io/register` to `[https://etherscan.io/register](https://etherscan.io/register)` please :)

adilanchian commented on NFT_Collection/en/Section_2/Lesson_2_Minting_NFTs_With_Static_SVGs.md:
````diff
@@ -112,6 +112,84 @@ Note: Remember to use `https://rinkeby.rarible.com/token/INSERT_DEPLOY_CONTRACT_
+We are down to our last step I promise. All that remains now is to run the command
+
+```
+npx hardhat verify YOUR_CONTRACT_ADDRESS --network rinkeby YOUR_CONSTRUCTOR_ARGUMENTS
+```
+
+Your constuctor arguments will simply be inital `_totalSupply` which you have set in your `run.js` script.
````
I think for this we can just put everything in the command like so:

`npx hardhat verify YOUR_CONTRACT_ADDRESS --network rinkeby _totalSupply`

chee-chyuan replied:
i was afraid it might not be obvious but i guess we can try first

adilanchian reviewed (changes requested):
Yessss we are getting there!! You're doing awesome :).

Going to tag @farzaa to get a second pair of eyes on this as well!

chee-chyuan commented:
thanks again. much cleaner now. im not really good in writing

adilanchian reviewed (approved):
Looks good to me! Thanks for working with me on making these changes :)

I think @farzaa wanted a few changes so will let him chime in to give approval as well

chee-chyuan commented:
> Looks good to me! Thanks for working with me on making these changes :)
> 
> I think @farzaa wanted a few changes so will let him chime in to give approval as well

lovely! just let me know of how i could make it better

adilanchian commented:
Thanks @chee-chyuan - will do!

farzaa commented:
Hey @chee-chyuan I thinki this should be in "finishing touches" under the github section. Would be a perfect fit there.

https://github.com/buildspace/buildspace-projects/blob/main/NFT_Collection/en/Section_4/Lesson_2_Ship_It.md

Also -- maybe at the top you can sorta explain why verifying is contract is cool and maybe link an example to one!! Showing off how you can hit the functions right in Etherscan -- and how magical that is!!! It's a cool feeling. Very different from github where you're required to pull the code down. Maybe even compare github to ethereum/etherscan. 

You want to motivate the user just a bit more to care!!!

Utkarshbhimte commented:
when I was doing this, It wasn't working till I removed all the console.logs, I think that is a useful side note to add. 
It gave me a lot of pain 😅

also adding to @farzaa's point, it is more essential to verify the contracts so you can ask for help easily in the community. Asking for help made it super simple when they were able to see what my code was clearly.

chee-chyuan commented:
> when I was doing this, It wasn't working till I removed all the console.logs, I think that is a useful side note to add. It gave me a lot of pain 😅
> 
> also adding to @farzaa's point, it is more essential to verify the contracts so you can ask for help easily in the community. Asking for help made it super simple when they were able to see what my code was clearly.

there shouldnt be a problem with console.log. in fact here is an example which i used with console.log https://rinkeby.etherscan.io/address/0x902ebbecafc54f7a8013a9d7954e7355309b50e6#code (refer file 5 of 14)

chee-chyuan commented:
hey @farzaa I have moved over the Verify Etherscan to the section 4 lesson 2. let me know if its good. also added some motivation which i think why its cool to verify contracts on Etherscan. my explanation will probably not be the best and do let me know if I can do anything to make it better.

Cheers

adilanchian commented on NFT_Collection/en/Section_2/Lesson_2_Minting_NFTs_With_Static_SVGs.md:
```diff
@@ -112,6 +112,7 @@ Note: Remember to use `https://rinkeby.rarible.com/token/INSERT_DEPLOY_CONTRACT_
 
 ![Untitled](https://i.imgur.com/Z2mKTpK.png)
 
+
```
Can you remove this new line please :)

adilanchian commented on NFT_Collection/en/Section_4/Lesson_2_Ship_It.md:
````diff
@@ -54,6 +54,83 @@ Remember, an NFT is just a JSON file at the end of the day that links to some me
+```
+
+Then in your `hardhat.config.js` add the following
+```
````
Can you change this to ```javascript please?

adilanchian reviewed (changes requested):
Looking good on my end after these changes!

adilanchian commented:
This is an insanely epic pr. Thanks again for making this! We are good to go woot woot!

adilanchian commented:
@all-contributors please add @chee-chyuan for doc

chee-chyuan commented:
> This is an insanely epic pr. Thanks again for making this! We are good to go woot woot!

what you guys are doing are way more epic!!! thanks for making buildspace happen. i only try to help out on the little stuff wherever i can. i will create a similar pr to add this feature into the nft game course
